controllers/vspheremachine_controller.go

```diff
@@ -181,8 +186,23 @@ func (r *VSphereMachineReconciler) reconcileVSphereMachineIPAddress(cluster *cap
 		//TODO: handle ipv6
 		//gateway4 is required if DHCP4 is disabled, gateway6 is required if DHCP6 is disabled
 		dev.Gateway4 = gateway
+
+		//if set, the list of nameservers and searchDomains should be retrieved from the IPPool
+		//if not, always default to the values from the VSphereMachineTemplate
 		dev.Nameservers = util.GetDNSServers(ipPool)
 		dev.SearchDomains = util.GetSearchDomains(ipPool)
+		if len(dev.Nameservers) == 0 || len(dev.SearchDomains) == 0 {
```

Comment:
with this the logic seems a bit not intuitive. what about we directly update the existing device[i], rather than create a new device object and then assign back? Then don't need to retrieve the MachineTemplate, because the Machine itself will be created from the template, so if template have nameserver/domain, Machine should already have it.

Reply:
so here we always update ip first, and only update nameserver/domain when ippool has it. don't need to check the machinetemplate.

Reply:
Ack, will update.